Simplified/NYPLAgeCheck.swift

```diff
@@ -18,12 +18,13 @@ import Foundation
   func verifyCurrentAccountAgeRequirement(_ completion: ((Bool) -> ())?) -> Void
   {
     serialQueue.async {
+      let userAccount = NYPLUserAccount.sharedAccount()
       guard let accountDetails = AccountsManager.shared.currentAccount?.details else {
         completion?(false)
         return
       }
       
-      if accountDetails.needsAuth == true || accountDetails.userAboveAgeLimit {
+      if userAccount.needsAuth == true || accountDetails.userAboveAgeLimit {
```

Comment:
yes, it is, as the result of `needsAuth` is based on authentication method. Previously it was Brough to the library level because it gave an output based on the first authentication method. As we do support multiple authentication methods now, we can't just check the type of the first one, and need to check this value for the selected one